### PR TITLE
Use Liquid::Expression::LITERALS to look up parsed literal names

### PR DESCRIPTION
**Pull request by pushrax**
@fw42 @dylanahsmith

### CODE REVIEW

File: ext/liquid_c/parser.c [hunk 1]
```diff
@@ -119,12 +119,10 @@ static VALUE parse_variable(parser_t *p)
...
     }
 
-    if (RARRAY_LEN(lookups) == 0 && TYPE(name) == T_STRING) {
-        if (rstring_eq(name, "nil") || rstring_eq(name, "null")) return Qnil;
-        if (rstring_eq(name, "true")) return Qtrue;
-        if (rstring_eq(name, "false")) return Qfalse;
-        if (rstring_eq(name, "blank")) return symBlank;
-        if (rstring_eq(name, "empty")) return symEmpty;
+    if (RARRAY_LEN(lookups) == 0) {
+        VALUE undefined = FIXNUM_P(-1);
```
Comment (pushrax) [inline ext/liquid_c/parser.c]: I had to do this since for some strange reason `rb_hash_has_key` is a static method...

**Comment by dylanahsmith** [comment]
LGTM

**Comment by fw42** [comment]
@csfrancis can you take a look? My Ruby C is weak

**Comment by fw42** [comment]
For context: This is to make https://github.com/Shopify/liquid/pull/592 work with liquid/c

File: ext/liquid_c/parser.c [hunk 2]
```diff
@@ -2,7 +2,7 @@
 #include "parser.h"
 #include "lexer.h"
 
-static VALUE cLiquidRangeLookup, cLiquidVariableLookup, cRange, symBlank, symEmpty;
+static VALUE cLiquidRangeLookup, cLiquidVariableLookup, cRange, vLiquidExpressionLiterals;
```
Comment (pushrax) [inline ext/liquid_c/parser.c]: What's the naming convention for values? I picked `v` arbitrarily, not sure what's commonly used.

File: ext/liquid_c/parser.c (hunk 1 above)
Comment (pushrax) [inline ext/liquid_c/parser.c]: There was a patch submitted in 2002 to change this haha. http://blade.nagaokaut.ac.jp/cgi-bin/scat.rb/ruby/ruby-core/86

**Comment by pushrax** [comment]
@trishume would be a good review as well, @csfrancis is super-busy

File: ext/liquid_c/parser.c [hunk 3]
```diff
@@ -194,5 +190,7 @@ void init_liquid_parser(void)
 
     VALUE cLiquidExpression = rb_const_get(mLiquid, rb_intern("Expression"));
     rb_define_singleton_method(cLiquidExpression, "c_parse", rb_parse_expression, 1);
+
+    vLiquidExpressionLiterals = rb_const_get(cLiquidExpression, rb_intern("LITERALS"));
```
Comment (csfrancis) [inline ext/liquid_c/parser.c]: You could maybe be defensive here and assert this is a hash.

**Comment by csfrancis** [comment]
LGTM.

**Comment by fw42** [comment]
Should we release a new gem version? Any non-backwards compatible changes since the last release?

**Pull request merged**